File: back-end/ProgettoFinale_IngegneriaDelSoftware/src/main/java/com/example/ProgettoFinale_IngegneriaDelSoftware/DTO/DipendenteDTO.java
package com.example.ProgettoFinale_IngegneriaDelSoftware.DTO;

import com.example.ProgettoFinale_IngegneriaDelSoftware.Composite.Composite.Gruppo;
import jakarta.persistence.Entity;
import lombok.*;

import java.util.List;

@Builder
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class DipendenteDTO {
    private Long id;
    private String nome;
    private String cognome;
    private String ruolo;
    private List<Gruppo> gruppoDTOList;
}

File: front-end/lib/model/DipendenteDTO.dart
class DipendenteDTO {
  
  String nome;
  String cognome;
  String ruolo;

  DipendenteDTO({required this.nome, required this.cognome, required this.ruolo});

  Map<String, dynamic> toJson() {
    return {
      'nome': nome,
      'cognome': cognome,
      'ruolo': ruolo,
    };
  }
}
